### PR TITLE
Build arm docker image

### PR DESCRIPTION
**Pull request by dbkr**
and a few other cleanups:

 * Use a builder image so the multiplatform build only has to compile once
 * Separate out dev dependencies

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,20 +11,22 @@
         "lint:js-fix": "eslint --ext .ts --fix ./src/*",
         "lint:types": "tsc --noEmit",
         "test:trafficlight": "rm -rf ./bin && tsc && node bin/src/index.js",
-        "docker-build": "rm -rf ./bin && tsc",
-        "docker-run": "node bin/src/index.js"
```
Comment (dbkr) [inline package.json:15]: These didn't really seem to have anything to do with docker and were just building / starting the project.

File: Dockerfile [hunk 2]
```diff
@@ -1,18 +1,17 @@
-FROM mcr.microsoft.com/playwright:v1.32.0-jammy
-
+FROM --platform=$BUILDPLATFORM mcr.microsoft.com/playwright:v1.32.0-jammy as builder
 WORKDIR /usr/src/app
 
-RUN mkdir /video
```
[the remark this answers is not in the document]
Comment (michaelkaye) [reply Dockerfile:5]: Ah, there's a folder called videos (which ends up in /usr/share/app/videos and a temporary working space that i mount outside the container called /videos. 

It's not unreasonable to remove this one; i started using usr/share/app/videos, but that had the side effect that the generated videos are mixed with the images folder and could be committed to git by accident. 

I need to refactor around those locations a little.

File: package.json (hunk 1 above)
Comment (michaelkaye) [reply package.json:15]: We should also make test:trafficlight just call yarn run build; yarn run start (which is where the split came from, so docker could pre-build)

File: Dockerfile [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
-COPY package*.json ./
-COPY tsconfig*.json ./
-
-RUN yarn install
+COPY package.json yarn.lock tsconfig.json trafficlight.config.json ./
```
[the remark this answers is not in the document]
Comment (michaelkaye) [reply Dockerfile:4]: Can we leave the `mkdir /video` in this PR at least; i've got a task to clean up where files are stored inside and outside docker myself.

**Review by michaelkaye — COMMENTED** [review]
I'll try the image locally, but i don't think video recording & upload will work as-is - some of the folder creation is still important. But it does need rearranging. 

The other changes look fine though.

**Comment by michaelkaye** [comment]
I think this change means we don't have weirdness with video being mounted around.
We split the image files that end up in ./bin (the source files) from  the temporary files that are used to power ffmpeg (in the given directory
We also store the snapshots etc in that video folder as well; overall we limit the amount of writable directories required to just that one folder (which is controllable via envvar or the config file).

File: Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (dbkr) [reply Dockerfile]: Oh I see: makes sense (although I would assume tsconfig & the trafficlight config would change fairly infrequently?)

**Review by dbkr — COMMENTED** [review]
Thanks - changes look sensible to me!

**Pull request merged**